src/hardware/mixer.cpp

```diff
@@ -2036,7 +2036,7 @@ bool MixerChannel::WakeUp()
 template <class Type, bool stereo, bool signeddata, bool nativeorder>
 void MixerChannel::AddSamples(const int num_frames, const Type* data)
 {
-	assert(num_frames >= 0);
+	assert(num_frames > 0);
```

Comment:
I'm pretty sure I ran into this assert triggering with 0 frames at some point.  It may have been while I was doing something dumb before I got my PR into the final state though so I guess we can leave it and see if it triggers again.

Reply:
Yeah, so general comment, it's extremely hard to 100% guarantee that audio devices will _never ever_ try to publish zero frames. And I don't really see why would that be a problem, either, it if happens occasionally.

One school of thought would be to make sure publishing zero frames works, but keep the `num_frames > 0` debug assert. I'm fine with that, it might draw attention to some bigger issues. Maybe. But I really want everything to handle zero-length data correctly in release builds.

Reply:
> One school of thought would be to make sure publishing zero frames works, but keep the num_frames > 0 debug asserts.

Yes; those are the couple adjustments I've made in the bulk dequeue; so it's safe in release builds but we can catch scenarios where non-blocking calls may be spinning hard with zero requests (because there's no back pressure to throttle them as they're non-blocking). At least we can be made aware and decide how to handle it.